src/compiler/checker.ts

```diff
@@ -31579,7 +31579,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
     function isExcludedMappedPropertyName(constraint: Type, propertyNameType: Type): boolean {
         if (constraint.flags & TypeFlags.Conditional) {
-            return !!(getReducedType(getTrueTypeFromConditionalType(constraint as ConditionalType)).flags & TypeFlags.Never) && isTypeAssignableTo(propertyNameType, (constraint as ConditionalType).extendsType);
+            const type = constraint as ConditionalType;
+            return !!(getReducedType(getTrueTypeFromConditionalType(type)).flags & TypeFlags.Never) &&
+                isTypeAssignableTo(getFalseTypeFromConditionalType(type), type.checkType) &&

```

Comment:
Actually, I think you want to check `===` here.

Reply:
I thought about it but then some weird conditionals like this one don't seem to break the rule. Or do they? 😅 
```ts
T extends "bar" ? never : T & ExtraFilter
```

Reply:
We have similar code in `getSimplifiedConditionalType` and there we check for identity. I'd prefer that.

Reply:
Ok, I pushed out a change that uses similar code to the one in `getSimplifiedConditionalType`